### ISSUE TITLE
Order less used actions from main context menu in submenus

### ISSUE THREAD

**Issue by teamcons** [elementary/files#2518]
### Problem

Less used actions, such as "Send in mail", create noise when looking for common actions such as "Move to trash" or "rename"

### Proposal

Identify a list of actions that could be moved into a "More actions" submenu, below of "Open with"

i may edit this FR once i have access to our eOS puter, but off of the top of my head:
-Send by email
-Create shortcut

This would make the file manager more focused, and in the same breath provide room for extending it

### Prior Art (Optional)

Nemo, in Cinnamon, has a "More tools" where user can expand with community-made "Actions"
Nemo and Nautilus have a scripts folder, allowing users to create custom actions on files via shell scripts

**Comment by jeremypw** [comment]
I agree the context menu for file items has become rather long. I think most items provided by contractors and plugins could be moved into a subfolder. These include "Send by Email", "Send by Bluetooth", "Compress", "Apply color tag", "Set as Desktop background/Wallpaper".   Incidentally I noticed there now seems to be two items that do the same thing (set the background) so that needs sorting out.  I'll raise a separate issue for that.

Flagging for design team input before implementing.

**Comment by teamcons** [comment]
Okay i think ill try my hand at this. But tbh doing a design seems to me much slower than doing the actual code and posting a screenshot of the branch in action
so ill try the latter

**Comment by jeremypw** [comment]
@teamcons I'll be glad to review code but it'll need sign off by @danirabbit .  Thanks for contributing to Code.

**Comment by teamcons** [comment]
No promise, im already struggling to locate the necessary snippet of code to amend

**Comment by jeremypw** [comment]
I would start by looking at the `show_context_menu` method of `AbstractDirectoryView`.  Plugins append their own items to this menu.

**Comment by danirabbit** [comment]
Couple things to keep in mind:

After the Gtk4 port we'll have a lot more layout freedom with popovers so some actions might be better done as image buttons which take up less space

I think in the future we probably will want some kind of portal-provided "share sheet" so that sandboxed apps can access things like "open with". It might make sense to think about which actions are specific to Files and which things should be available too any app sharing/exporting data and separate those things into a new "sharing" dialog to kind of prototype that. There's some prior art from other platforms in the portals repo I think

I think the way the context menu is built right now relies on GTK.menuitem pretty heavily so generally this will all have to be rewritten to use actions for the gtk4 port. So any work here should probably make sure it's not making that port even harder

**Comment by jeremypw** [comment]
One of the tasks on the porting roadmap is to convert menus to GLib.Menu but you may feel that is outside the scope of this issue.

**Comment by teamcons** [comment]
okay better to wait then

**Comment by jeremypw** [comment]
I feel reorganising/rewriting existing menus is a task possibly with sub-tasks or a project rather than a new feature.

**Comment by jeremypw** [comment]
#1179 is a sub-task of this (although contractor items are likely to migrate to portal or internal items)

**Comment by teamcons** [comment]
If this was gtk4 i would have tried my hands, but right now this feels like it would be a waste of effort
And i can help with _some_ of the porting, but this is a too huge and complicated project to do more than a couple easy stuff

**Comment by jeremypw** [comment]
@teamcons Thanks for the offer 😄 We are still at the preparation stage atm so there is still a lot of fairly straightforward code cleanup to do before starting the port. See the prep PRs dani has done for may other projects.
